src/hashing.rs

```diff
@@ -227,8 +231,18 @@ pub struct DomainSeparatedHasher<D, M> {
 }
 
 impl<D: Digest, M: DomainSeparation> DomainSeparatedHasher<D, M> {
+    /// Create a new instance of [`DomainSeparatedHasher`] without an additional label (to correspond to 'D::new()').
+    pub fn new() -> Self {
```

Comment:
This is not correct. It does not call `M::add_domain_separation_tag`
```suggestion
    pub fn new() -> Self {
       Self::new_with_label("")
```